arviz/plots/distcomparisonplot.py

```diff
@@ -134,6 +135,22 @@ def plot_dist_comparison(
         for data, var in zip(datasets, var_names)
     ]
 
+    total_plots = sum(
+        sum(1 for _ in xarray_var_iter(data, var_names=var, combined=True))
+        for data, var in zip(datasets, var_names)
+    )
```

Comment:
Here similar comment as for next line, the nested loop is not needed and is actually incorrect. It should be something like:

    total_plots = sum(1 for _ in xarray_sel_iter(datasets[0], var_names=var_names[0], combined=True))

Also as we only want to count, we can use `sel` instead of using `var`, we don't need to subset the data to get the values in each position to count how many positions there are.

Reply:
Okay thank you for this explanation it was really useful and sorry for all the trouble caused, it's my first time contributing.  
So what I have understood and already done (no commit yet) is that I have compared the `total_plots` which are the desired plots by the user with the `rcParams["plot.max_subplots"]`. If `total_plots` are greater than this then the warning is popping up and it does not truncate the plots because they are already truncated at the creation. I also used the `maxplots` only at the description of the warning. I've also included a test case.

Reply:
Sounds great, and don't worry about the discussion and back-and-forth comments, I think they were useful to all of us.